OpenRA.Game/Network/Nat.cs

```diff
@@ -0,0 +1,102 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2021 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Threading;
+using Mono.Nat;
+
+namespace OpenRA.Network
+{
+	public enum NatStatus { Enabled, Disabled, NotSupported }
+
+	public class Nat
+	{
+		public static NatStatus Status
+		{
+			get
+			{
+				return NatUtility.IsSearching ? natDevice != null ?
+					NatStatus.Enabled : NatStatus.NotSupported : NatStatus.Disabled;
+			}
+		}
+
+		static Mapping mapping;
+		static INatDevice natDevice;
+
+		public static void Discover()
```

Comment:
Minor nit: IMO `Initialize` would be a clearer name here, or perhaps `InitializeAndDiscover` if we want to be really explicit.

We should have a `bool initialized` guard to make sure we can't accidentally run this more than once.